Bootloader/Adapters/Src/hash_adapter.c

```diff
@@ -0,0 +1,96 @@
+ *   Authors: Igor Misic <igy1000mb@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name IMProject nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "hash_adapter.h"
+#include "monocypher.h"
+#include "base64.h"
+#include "main.h"
+
+
+void
+HashAdapter_getHashedBoardId(uint8_t hashed_board_id[HASHED_BOARD_ID_SIZE]) {
+
+    uint32_t uid[3] = {0U};
+    uid[0] = HAL_GetUIDw0();
+    uid[1] = HAL_GetUIDw1();
+    uid[2] = HAL_GetUIDw2();
+
+    uint8_t manufacturer_id[MANUFACTURER_ID_SIZE];
+
+    if(BoardInfo_decodeBase64ManufacturerId(manufacturer_id)) {
+
+        switch (HASH_BOARD_ID_ALGORITHM) {
+
+            case (BLAKE2B):
+                crypto_blake2b_general(hashed_board_id, HASHED_BOARD_ID_SIZE, manufacturer_id, MANUFACTURER_ID_SIZE, (uint8_t*)uid, 3 * sizeof(uint32_t));
+                break;
+
+            case (SHA256):
+                //Only SMT32H75x supports HW accelerator
+                break;
+            default:
+                break;
+        }
+    }
+}
+
+bool
+HashAdapter_getBase64HashedBoardId(uint8_t* b64_hashed_board_id) {
+
+    bool success = false;
+
+    uint8_t hashed_board_id[HASHED_BOARD_ID_SIZE] = {0U};
+    HashAdapter_getHashedBoardId(hashed_board_id);
+    int32_t ret = Base64_encode(hashed_board_id, HASHED_BOARD_ID_SIZE, (char*)b64_hashed_board_id, BASE64_HASHED_BOARD_ID_SIZE);
+
+    if (0 == ret) {
+        success = true;
+    }
+
+    return success;

```

Comment:
success variable is not needed.

return (ret == 0);

Reply:
This is intentionally here. I want to have a code style where we return variables.
So in this case each function that returns boolean as success should have this template:

```
bool function_name()
{
  bool success = false; //or true
  return success;
}
```